frame/contracts/src/lib.rs

```diff
@@ -724,15 +732,24 @@ pub mod pallet {
 			)
 		}
 
-		/// TODO: add benchmark for base weight
-		/// TODO: add a minimum weight limit so that people don't spam small free extrinsics
```

Comment:
picked an arbitrary min weight_limit here, should we make that configurable through the Config?

Reply:
My thought was that if a migration takes thousands of steps (e.g when we migrate one contract at a time) then asking for at least 10x was reasonable. That was my reading of  your comment "add a minimum weight limit so that people don't spam small free extrinsic"

Reply:
went with your suggestion, 
> For this we probably need to return the number of steps performed from MigrateSequence::steps

I don't think that's necessary the Migration::<T>::migrate returns a MigrateResult that tells you wether or not No migration was performed

Reply:
Only if it errors out before going into `steps`. You could just supply enough weight to be able to enter `steps` but not enough for a single iteration of its loop. AFAIK this will then result in a free transaction.